app/forms/questionnaire_form.py

```diff
@@ -377,21 +374,43 @@ def _get_error_id(answer_id):
     return f"{answer_id}-error"
 
 
+def _clear_detail_answer_field(form_data, question):
+    """
+    Checks the submitted answers and in the case of both checkboxes and radios,
```

Comment:
I found the comment a bit hard to follow and a little verbose. Isn't it just "clear detail_answer if parent option not selected"? Also I am not sure we need the params, nothing else in that file states them and they were obvious to me